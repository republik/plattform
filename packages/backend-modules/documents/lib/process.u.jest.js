jest.mock('@orbiting/backend-modules-auth', () => ({
  Roles: {
    userIsInRoles: jest.fn(),
  },
}))

const UUT = require('./process')

const getMdast = () => ({
  children: [
    {
      identifier: 'FIGURE',
      data: {},
      children: [
        {
          children: [
            {
              alt: null,
              type: 'image',
              title: null,
              url: '',
            },
          ],
          type: 'paragraph',
        },
        {
          children: [
            {
              children: [
                {
                  type: 'text',
                  value: 'P',
                },
              ],
              type: 'emphasis',
            },
          ],
          type: 'paragraph',
        },
      ],
      type: 'zone',
    },
    {
      identifier: 'CENTER',
      data: {},
      children: [
        {
          children: [
            {
              type: 'text',
              value: 'Paragraph 1',
            },
          ],
          type: 'paragraph',
        },
        {
          identifier: 'IF',
          data: {
            present: 'hasAccess',
          },
          children: [
            {
              children: [
                {
                  type: 'text',
                  value: 'If 2-1',
                },
              ],
              type: 'paragraph',
            },
            {
              children: [
                {
                  type: 'text',
                  value: 'If 2-2',
                },
              ],
              type: 'paragraph',
            },
            {
              identifier: 'ELSE',
              data: {},
              children: [
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 2-1',
                    },
                  ],
                  type: 'paragraph',
                },
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 2-2',
                    },
                  ],
                  type: 'paragraph',
                },
              ],
              type: 'zone',
            },
          ],
          type: 'zone',
        },
        {
          children: [
            {
              type: 'text',
              value: 'Paragraph 2',
            },
          ],
          type: 'paragraph',
        },
        {
          identifier: 'IF',
          data: {
            present: 'hasAccess',
          },
          children: [
            {
              identifier: 'ELSE',
              data: {},
              children: [
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 3-1',
                    },
                  ],
                  type: 'paragraph',
                },
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 3-2',
                    },
                  ],
                  type: 'paragraph',
                },
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 3-3',
                    },
                  ],
                  type: 'paragraph',
                },
              ],
              type: 'zone',
            },
          ],
          type: 'zone',
        },
        {
          identifier: 'IF',
          data: {
            present: 'hasAccess',
          },
          children: [
            {
              children: [
                {
                  type: 'text',
                  value: 'If 4-1',
                },
              ],
              type: 'paragraph',
            },
            {
              identifier: 'ELSE',
              data: {},
              children: [
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 4-1',
                    },
                  ],
                  type: 'paragraph',
                },
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Else 4-2',
                    },
                  ],
                  type: 'paragraph',
                },
                {
                  identifier: 'IF',
                  data: {
                    present: 'hasAccess',
                  },
                  children: [
                    {
                      children: [
                        {
                          type: 'text',
                          value: 'If 5-1',
                        },
                      ],
                      type: 'paragraph',
                    },
                    {
                      children: [
                        {
                          type: 'text',
                          value: 'If 5-2',
                        },
                      ],
                      type: 'paragraph',
                    },
                    {
                      children: [
                        {
                          type: 'text',
                          value: 'If 5-3',
                        },
                      ],
                      type: 'paragraph',
                    },
                    {
                      identifier: 'ELSE',
                      data: {},
                      children: [
                        {
                          children: [
                            {
                              type: 'text',
                              value: 'Else 5-1',
                            },
                          ],
                          type: 'paragraph',
                        },
                        {
                          children: [
                            {
                              type: 'text',
                              value: 'Else 5-2',
                            },
                          ],
                          type: 'paragraph',
                        },
                      ],
                      type: 'zone',
                    },
                  ],
                  type: 'zone',
                },
              ],
              type: 'zone',
            },
          ],
          type: 'zone',
        },
        {
          children: [
            {
              type: 'text',
              value: 'Paragraph 6',
            },
          ],
          type: 'paragraph',
        },
        {
          identifier: 'IF',
          data: {
            present: 'hasAccess',
          },
          children: [
            {
              children: [
                {
                  type: 'text',
                  value: 'If 7-1',
                },
              ],
              type: 'paragraph',
            },
            {
              children: [
                {
                  type: 'text',
                  value: 'If 7-2',
                },
              ],
              type: 'paragraph',
            },
          ],
          type: 'zone',
        },
        {
          children: [
            {
              type: 'text',
              value: 'Paragraph 8',
            },
          ],
          type: 'paragraph',
        },
      ],
      type: 'zone',
    },
  ],
  type: 'root',
})

describe('process', () => {
  describe('processIfHasAccess', () => {
    it('keeps non-member nodes if user has no access', () => {
      const { Roles } = require('@orbiting/backend-modules-auth')
      Roles.userIsInRoles.mockReturnValue(false)

      const mdast = getMdast()

      UUT.processIfHasAccess(mdast)
      expect(mdast).toMatchSnapshot()
    })

    it('keeps member nodes if user has access', () => {
      const { Roles } = require('@orbiting/backend-modules-auth')
      Roles.userIsInRoles.mockReturnValue(true)

      const mdast = getMdast()
      UUT.processIfHasAccess(mdast)
      expect(mdast).toMatchSnapshot()
    })
  })
})
