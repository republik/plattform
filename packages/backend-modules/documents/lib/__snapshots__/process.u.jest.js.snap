// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process processIfHasAccess keeps member nodes if user has access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "alt": null,
              "title": null,
              "type": "image",
              "url": "",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "P",
                },
              ],
              "type": "emphasis",
            },
          ],
          "type": "paragraph",
        },
      ],
      "data": Object {},
      "identifier": "FIGURE",
      "type": "zone",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "If 2-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "If 2-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "If 4-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 6",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "If 7-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "If 7-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 8",
            },
          ],
          "type": "paragraph",
        },
      ],
      "data": Object {},
      "identifier": "CENTER",
      "type": "zone",
    },
  ],
  "type": "root",
}
`;

exports[`process processIfHasAccess keeps non-member nodes if user has no access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "alt": null,
              "title": null,
              "type": "image",
              "url": "",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "P",
                },
              ],
              "type": "emphasis",
            },
          ],
          "type": "paragraph",
        },
      ],
      "data": Object {},
      "identifier": "FIGURE",
      "type": "zone",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 2-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 2-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 3-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 3-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 3-3",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 4-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 4-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 5-1",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Else 5-2",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 6",
            },
          ],
          "type": "paragraph",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Paragraph 8",
            },
          ],
          "type": "paragraph",
        },
      ],
      "data": Object {},
      "identifier": "CENTER",
      "type": "zone",
    },
  ],
  "type": "root",
}
`;
